.github/workflows/build-container.yml

```diff
@@ -105,28 +109,41 @@ jobs:
           exit 1
         fi
     - name: Deploy to cluster
-      run:
-        kubectl apply -f e2e-tests/kubernetes-headlamp-ci.yaml
+      run: kubectl apply -f e2e-tests/kubernetes-headlamp-ci.yaml
     - name: Run e2e tests
       run: |
-        echo "------------------sleeping 3...------------------"
+        echo "------------------------------------sleeping 3...------------------------------------"
         sleep 6
-        minikube service list
-        minikube service headlamp -n kube-system --url
+        kubectl get services --all-namespaces
         kubectl get deployments -n kube-system
-        minikube logs | tail -10
-        echo "------------------waiting for headlamp deployment to be available...------------------"
+        echo "------------------Waiting for headlamp deployment to be available...------------------"
         kubectl wait deployment -n kube-system headlamp --for condition=Available=True --timeout=30s
-        minikube service headlamp -n kube-system --url
-        echo "------------------opening the service------------------"
-        export SERVICE_URL=$(minikube service headlamp -n kube-system --url | tail -1)
+        echo "----------------------------------Opening the service----------------------------------"
+        IP_ADDRESS=$(kubectl get nodes -o=jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}')
+        SERVICE_PORT=$(kubectl get services headlamp -n kube-system -o=jsonpath='{.spec.ports[0].nodePort}')
+        export SERVICE_URL="http://${IP_ADDRESS}:${SERVICE_PORT}"
         echo $SERVICE_URL
         curl -L $SERVICE_URL | grep -q "Headlamp: Kubernetes Web UI"        
-        echo "------------------Getting HEADLAMP_TOKEN------------------"
+        echo "--------------------------------Getting HEADLAMP_TOKEN--------------------------------"
         kubectl create serviceaccount headlamp-admin --namespace kube-system
         kubectl create clusterrolebinding headlamp-admin --serviceaccount=kube-system:headlamp-admin --clusterrole=cluster-admin
         export HEADLAMP_TOKEN=$(kubectl create token headlamp-admin --duration 24h -n kube-system)
-        echo "------------------Running playwright e2e tests------------------"
+        echo "---------------------------------Certificate handling---------------------------------"
+        export KUBECONFIG=$HOME/.kube/config
```

Comment:
I don't understand this really... Are there any documentation links for this you know of?

Reply:
The issue is that the kind kubeconfig does not specify local paths for the certificates, and our e2e test setup tries to read these paths. I copied the certificate handling over from the k3d setup but I suppose there should be a simpler way to authenticate, I'll continue testing